### PR TITLE
fix(video.js): add missing player's options and methods

### PR DESCRIPTION
**Pull request by nonth**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - Player's options
    - `autoplay` https://videojs.com/guides/options/#autoplay
    - `audioOnlyMode` https://videojs.com/guides/options/#audioonlymode
    - `audioPosterMode` https://videojs.com/guides/options/#audiopostermode
    - `autoSetup` https://videojs.com/guides/options/#autosetup
    - `breakpoints` https://videojs.com/guides/options/#breakpoints
    - `fullscreen` https://videojs.com/guides/options/#fullscreen
    - `id` https://videojs.com/guides/options/#id
    - `liveTracker`
      - https://videojs.com/guides/options/#livetrackertrackingthreshold
      - https://videojs.com/guides/options/#livetrackerlivetolerance
    - `normalizeAutoplay` https://videojs.com/guides/options/#normalizeautoplay
    - `preferFullWindow` https://videojs.com/guides/options/#preferfullwindow
    - `restoreEl` https://videojs.com/guides/options/#restoreel
    - `suppressNotSupportedError` https://videojs.com/guides/options/#suppressnotsupportederror
    - `techCanOverridePoster` https://videojs.com/guides/options/#techcanoverrideposter
    - `vtt.js` https://videojs.com/guides/options/#vttjs
    - `playsinline` https://github.com/videojs/video.js/blob/main/src/js/player.js#L1170
    - `disablePictureInPicture` https://github.com/videojs/video.js/blob/main/src/js/player.js#L1173
    - `enableSourceset` https://github.com/videojs/video.js/blob/main/src/js/player.js#L1180
    - `retryOnError` https://github.com/videojs/video.js/blob/main/src/js/player.js#L3464
  - Player's methods
    - `audioOnlyMode` https://github.com/videojs/video.js/blob/main/src/js/player.js#L4370
    - `audioPosterMode`https://github.com/videojs/video.js/blob/main/src/js/player.js#L4447
    - `breakpoints` https://github.com/videojs/video.js/blob/main/src/js/player.js#L4800
    - `crossOrigin` https://github.com/videojs/video.js/blob/main/src/js/player.js#L849
    - `debug` https://github.com/videojs/video.js/blob/main/src/js/player.js#L5097
    - `disablePictureInPicture` https://github.com/videojs/video.js/blob/main/src/js/player.js#L3074
    - `isInPictureInPicture` https://github.com/videojs/video.js/blob/main/src/js/player.js#L3094

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## video.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/video.js@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'video.js'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. hooks_
1. hooks
1. hook
1. hookOnce

<details>
<summary>as well as these 52 other properties...</summary>
<p>removeHook, VERSION, options, getPlayers, getPlayer, getAllPlayers, players, getComponent, registerComponent, getTech, registerTech, use, middleware, browser, TOUCH_ENABLED, extend, mergeOptions, bind, registerPlugin, deregisterPlugin, plugin, getPlugins, getPlugin, getPluginVersion, addLanguage, log, createLogger, createTimeRange, createTimeRanges, formatTime, setFormatTime, resetFormatTime, parseUrl, isCrossOrigin, EventTarget, on, one, off, trigger, xhr, TextTrack, AudioTrack, VideoTrack, computedStyle, dom, url, mediaSources, MediaSource, URL, HlsHandler, HlsSourceHandler, Hls</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a5d9d8b43d1afd51b1df0f2daf6499a67fc0b303
</p>

File: types/video.js/index.d.ts [hunk 1]
```diff
@@ -6195,6 +6207,35 @@ export interface VideoJsPlayer extends videojs.Component {
...
+     * Setting this to `true` will hide all player components except the control bar,
+     * as well as control bar components needed only for video.
+     *
+     * @param [value]
+     *         The value to set audioOnlyMode to.
+     *
+     * @return A Promise is returned when setting the state, and a boolean when getting
+     *         the present state
+     */
+    audioOnlyMode(value: boolean): void;
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:6222]:
```suggestion
    audioOnlyMode(value: boolean): Promise<boolean>;
```
https://github.com/videojs/video.js/pull/7647/files#diff-b62e95b3e474754b0b665ff01617843398a03815683899a9c79f141bab6a1fa7R2825

File: types/video.js/index.d.ts [hunk 2]
```diff
@@ -6195,6 +6207,35 @@ export interface VideoJsPlayer extends videojs.Component {
...
+
+    audioOnlyMode(): Promise<void> | boolean;
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:6224]:
```suggestion
    audioOnlyMode(): boolean;
```
https://github.com/videojs/video.js/pull/7647/files#diff-b62e95b3e474754b0b665ff01617843398a03815683899a9c79f141bab6a1fa7R2851

File: types/video.js/index.d.ts [hunk 3]
```diff
@@ -6195,6 +6207,35 @@ export interface VideoJsPlayer extends videojs.Component {
...
+     *
+     * @param [value]
+     *         The value to set audioPosterMode to.
+     *
+     * @return A Promise is returned when setting the state, and a boolean when getting
+     *         the present state
+     */
+    audioPosterMode(value: boolean): void;
+
+    audioPosterMode(): Promise<void> | boolean;
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:6237]:
```suggestion
    audioPosterMode(value: boolean): Promise<void>
    audioPosterMode(): boolean;
```
https://github.com/videojs/video.js/blob/3ec2ac7f99704434d2fdc61642247ef0aa4e77b3/test/unit/player.test.js#L1608-L1612

File: types/video.js/index.d.ts [hunk 4]
```diff
@@ -7111,11 +7237,11 @@ export interface VideoJsPlayer extends videojs.Component {
     width(value: number): void;
 
     width(): number;
-}
+  }
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:7240]:
```suggestion
}
```

File: types/video.js/index.d.ts [hunk 5]
```diff
@@ -6271,6 +6311,45 @@ export interface VideoJsPlayer extends videojs.Component {
...
+     *
+     * @param  [breakpoints.xlarge]
+     *         The maximum width for the "vjs-layout-x-large" class.
+     *
+     * @param  [breakpoints.huge]
+     *         The maximum width for the "vjs-layout-huge" class.
+     *
+     * @return An object mapping breakpoint names to maximum width values.
+     */
+    breakpoints(breakpoints: Partial<videojs.Breakpoint>): void;
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:6349]:
```suggestion
    breakpoints(breakpoints: true | Partial<videojs.Breakpoint>): void;
```
>      ...      If `true` is given, will set default breakpoints.

File: types/video.js/index.d.ts [hunk 6]
```diff
@@ -6352,6 +6431,22 @@ export interface VideoJsPlayer extends videojs.Component {
...
+     * behavior.
+     *
+     * @param [value]
+     *        The value to set the `Player`'s crossOrigin to. If an argument is
+     *        given, must be one of `anonymous` or `use-credentials`.
+     *
+     * @return - The current crossOrigin value of the `Player` when getting.
+     *         - undefined when setting
+     */
+    crossOrigin(value: string): void;
```
Comment (peterblazejewicz) [inline types/video.js/index.d.ts:6446]:
```suggestion
    crossOrigin(value: 'anonymous' | 'use-credentials'): void;
```
>  If an argument is  given, must be one of `anonymous` or `use-credentials`.

Comment (peterblazejewicz) [reply types/video.js/index.d.ts:6446]: you can make into type (string literals union), to keep convention for those types.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@nonth thanks!

**Comment by nonth** [comment]
Ready to merge

**Comment by nonth** [comment]
@peterblazejewicz Thanks for approved my pr.

Could you please take a look on this discussion https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/61426 if it ok I can create another pr for it

**Pull request merged**